src/main/java/org/mockito/internal/configuration/injection/ConstructorInjection.java

```diff
@@ -58,7 +55,7 @@ public boolean processInjection(Field field, Object fieldOwner, Set<Object> mock
     }
 
     /**
-     * Returns mocks that match the argument type, if not possible assigns null.
+     * Returns mocks that match the argument type, if not possible tries to assign an anonymous mock or defaults to null.
```

Comment:
Can you pls explain on assigning an anonymous mock would help , especially when there is no mock for that argument type ?